src/create-config.ts

```diff
@@ -31,7 +13,7 @@ export type CreateStencilPlaywrightConfigOptions = Partial<PlaywrightTestConfig>
  * @param overrides Values to override in the default config. Any Playwright config option can be overridden.
  * @returns A {@link PlaywrightTestConfig} object
  */
-export const createConfig = async (overrides?: CreateStencilPlaywrightConfigOptions): Promise<PlaywrightTestConfig> => {
+export const createConfig = async (overrides?: Partial<PlaywrightTestConfig>): Promise<PlaywrightTestConfig> => {
```

Comment:
I just tried this out, and I think we might need to adjust the typing just a little bit. I tried out the following in a sample project:

```ts
import { expect } from '@playwright/test';
import { matchers, createConfig } from '@stencil/playwright';

// Add custom Stencil matchers to Playwright assertions
expect.extend(matchers);

export default createConfig({
  webServer: {
    port: 4444
  }
});
```

I think based on what we're doing here this should be ok, but I get a type error because in the `webServer` object the `command` property isn't optional:

https://github.com/microsoft/playwright/blob/1ef85015f3f389d7b968792a39bf6e0550cc1821/packages/playwright/types/test.d.ts#L8298-L8355

this shouldn't matter to us since we're going to supply a default value anyway.

But I think to fix this we just need to make the `Partial` recursive so that it doesn't just apply to top-level properties on `PlaywrightTestConfig`, I think something like this would work:

```ts
type DeepPartial<T> = T extends Object ? {
  [K in keyof T]+?: DeepPartial<T[K]>
} : T;
```

then if the argument to `createConfig` is `DeepPartial<PlaywrightTestConfig>` I think the user can be free to specify any option at any level - I tested it locally a bit and it got rid of the error I saw when only specifying `webServer.port` for instance

Reply:
Oh dang, great catch!